src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const container = [];
+  let clones = Object.assign({}, state);
+
+  for (const act in actions) {
+    const operation = actions[act];
+    const newClones = Object.assign({}, clones);
+
+    switch (operation.type) {
+      case 'addProperties':
+        container.push(Object.assign(newClones, operation.extraData));
```

Comment:
You can push only once on each iteration it is unneccessary to duplicate this code